src/api/mutable_data_permissions.js

```diff
@@ -54,63 +52,42 @@ module.exports.getPermissionsSet = (permissionsHandle, signKeyHandle) => {
  *
  * @param {PermissionsHandle} permissionsHandle the Permissions handle
  * @param {SignKeyHandle|null} signKeyHandle the sign key to map to
- * @param {PermissionsSetHandle} pmSetHandle - the permissions set you'd like insert
+ * @param {Array} permSetArray - array of permissions
  *
  * @returns {Promise} resolves once finished
  *
- * @example // Inserting a new permissions set into a MutableData:
- * let pmSetHandle, appSignKeyHandle, permsHandle;
- * window.safeCrypto.getAppPubSignKey(appHandle)
- *    .then((pk) => appSignKeyHandle = pk)
- *    .then(_ => window.safeMutableData.getPermissions(mdHandle))
- *    .then((h) => permsHandle = h)
- *    .then(_ => window.safeMutableData.newPermissionSet(appHandle))
- *    .then((h) => pmSetHandle = h)
- *    .then(_ => window.safeMutableDataPermissionsSet.setAllow(pmSetHandle, 'Insert'))
- *    .then(_ => window.safeMutableDataPermissionsSet.setAllow(pmSetHandle, 'ManagePermissions'))
- *    .then(_ => window.safeMutableDataPermissions.insertPermissionsSet(permsHandle, appSignKeyHandle, pmSetHandle))
- *    .then(_ => console.log('Finished inserting new permissions'));
- **/
-module.exports.insertPermissionsSet = (permissionsHandle, signKeyHandle, pmSetHandle) => {
+ */
+module.exports.insertPermissionsSet = (permissionsHandle, signKeyHandle, permSetArray) => {
   return getObj(signKeyHandle, true)
-    .then((signKeyObj) => getObj(pmSetHandle)
-      .then((pmSetObj) => getObj(permissionsHandle)
-        .then((permsObj) => permsObj.netObj.insertPermissionSet(signKeyObj.netObj, pmSetObj.netObj))
-      ));
+  .then((signKeyObj) => getObj(permissionsHandle)
+    .then((permsObj) => permsObj.netObj.insertPermissionSet(signKeyObj.netObj, permSetArray))
+  );
 };
 
 /**
- * Iterate over the entries, execute the function every time
- * @name window.safeMutableDataPermissions.forEach
+ * Return list of all associated permission-sets
+ * @name window.safeMutableDataPermissions.listPermissionSets
  *
  * @param {PermissionsHandle} permissionsHandle the Permissions handle
- * @param {function(Buffer, ValueVersion)} fn the function to call
  *
- * @returns {Promise} resolves once the iteration is finished
+ * @returns {Promise<Array>} resolves once the iteration is finished
  *
  * @example // Iterating over the permissions of a MutableData:
  * window.safeMutableData.getPermissions(mdHandle)
- *    .then((permsHandle) => window.safeMutableDataPermissions.forEach(permsHandle, (p) => {
- *          console.log('Permissions entry handle: ', p);
- *       }).then(_ => console.log('Iteration finished'))
- *    );
- **/
-module.exports._with_cb_forEach = (permissionsHandle) => {
-  return forEachHelper(permissionsHandle, true);
+ *    .then((permsHandle) => window.safeMutableDataPermissions.listPermissionSets(permsHandle))
+ *    .then((permsArray) => {
+ *      console.log(permsArray);
```

Comment:
Please put some text ion the `console.log` just to make the example nicer/cleaner, e.g. console.log("Permission Sets retrieved:", permsArray);